### ISSUE TITLE
John Palsberg

### ISSUE THREAD

**Issue by johnpalsberg** [cucapra/undergrad-research#130]
## Personal Details

**Name:** John Palsberg

**Undergrad or MEng?** Undergrad

**Year in Cornell:** Freshman
 

**Relevant classes:** CS 2110, CS 3110, CS 2800, Math 2210

**Expertise (languages/frameworks/etc.):** Java, Python, C++, JavaScript, Gradle, git, SQL, Unix, bash, IntelliJ IDEA, Eclipse, Visual Studio Code.

## Research

**When do you want to do research?**
 I am interested in summer research, as well as extracurricular research during the semester. 

**What is exciting to you about research?**
 I love to learn new things and creatively generate new ideas, and I think that research is one of the best ways to fulfill those interests.

**What kind of research do you want to do?**
  I'd like to do research on the interaction between hardware and software. 

## Background

**Was there a paper that particularly excited you?**
 The paper "Static Stages for Heterogeneous Programming" was particularly interesting for me because I am fascinated by CPU-GPU coordination in computing.

**Which of the [current research projects](https://capra.cs.cornell.edu/) would you be interested in working on and why?** I am interested in Filament, as I am interested in HDLs, and Braid, since I am interested in scheduling hardware resources with software.

**Anything else you want to tell us about yourself?** 
In CS, I am specifically interested in topics focusing on low-level programming and working with hardware. During my senior year of high school, I along with a friend programmed a RISC-V simulator in Java, and then programmed a rudimentary operating system for scheduling multiple programs at once on the simulator. I then programmed algorithms in assembly to run on the simulator. This project further stimulated my interest in low-level computing, and I hope to explore even more by doing research!

**Attach a CV/Resumé:**
[John Palsberg Resume.pdf](https://github.com/cucapra/undergrad-research/files/14227953/John.Palsberg.Resume.pdf)

**Comment by sampsyo** [comment]
Let's keep this open in case something makes sense in the fall!

**Comment by johnpalsberg** [comment]
Thank you! I'm definitely still interested for this fall.

**Comment by sampsyo** [comment]
Hi, @johnpalsberg—I imagine it's quite likely that you've found other interesting research to do in the intervening time, but I'm starting to look for some collaborators to work on FlatGFA stuff (I think you saw a talk about this over the summer). Would you have any interest in that? Absolutely no worries if that's what what you're into; I bet there will be other opportunities sometime soon as well.

**Comment by johnpalsberg** [comment]
Hi Professor Sampson, thank you so much! I do remember your talk on the FlatGFA project this summer, and I am definitely interested in joining the project. Currently, I'm working with Professor White in a continuation of our BURE project, which will end by the end of this semester. What might onboarding into the group look like?

**Comment by johnpalsberg** [comment]
@sampsyo

**Comment by sampsyo** [comment]
Oh cool. Maybe it makes sense to pursue this in the spring in that case, if you're focusing on that games project for now? It's pretty flexible; I'm sure there will be more to do in the spring.

For whatever it's worth, I originally put together [some notes](https://capra.cs.cornell.edu/notes/pollen-meng.html) on what we'd be looking for as an MEng project (nobody was interested 🙃). That might also give you some idea of where we're going, even if you just totally ignore the MEng details.

**Comment by johnpalsberg** [comment]
Yeah, starting in the spring would be great. Thanks for the additional resources.

**Comment by johnpalsberg** [comment]
Hi @sampsyo,
I'm looking forward to joining the FlatGFA project soon. Has the schedule for the group meetings been decided yet? I'd like to know if there are any specific times in my schedule that I should leave free. 
Thanks!

**Comment by sampsyo** [comment]
Closing because @johnpalsberg is onboarded to work on Pollen stuff!
